tests/imap/test_save_folder_names.py

```diff
@@ -19,7 +19,7 @@ def test_imap_save_generic_folder_names(db, default_account):
     raw_folders = [RawFolder(*args) for args in folder_names_and_roles]
     monitor.save_folder_names(db.session, raw_folders)
     saved_folder_data = set(
-        db.session.query(Folder.name, Folder.canonical_name).filter(
+        db.session.query(Folder.name, Folder._canonical_name).filter(
```

Comment:
Do you still need to change these to ```_canonical_name``` with the extra conditional in the ```canonical_name``` setter? I think the purpose of ```_canonical_name = Column(String(MAX_INDEXABLE_LENGTH), nullable=False, default='', name="canonical_name")``` (in folder.py, https://github.com/nylas/sync-engine/pull/370/files) was that the "name" attribute would allow us to still query with ```canonical_name```. There seems to be plenty of other places in the codebase that use Folder.canonical_name, so I'm wondering if we need to change all of them, or if none of them need to be changed. (If we need to change all of them, I should revisit that PR)